query-engine/connectors/sql-query-connector/src/row.rs

```diff
@@ -230,13 +230,46 @@ fn row_value_to_prisma_value(p_value: Value, meta: ColumnMetadata<'_>) -> Result
             }
             _ => return Err(create_error(&p_value)),
         },
-        TypeIdentifier::Int | TypeIdentifier::BigInt => match p_value {
+        TypeIdentifier::Int => match p_value {
+            value if value.is_null() => PrismaValue::Null,
             Value::Int32(Some(i)) => PrismaValue::Int(i as i64),
             Value::Int64(Some(i)) => PrismaValue::Int(i),
             Value::Bytes(Some(bytes)) => PrismaValue::Int(interpret_bytes_as_i64(&bytes)),
             Value::Text(Some(ref txt)) => {
                 PrismaValue::Int(i64::from_str(txt.trim_start_matches('\0')).map_err(|_| create_error(&p_value))?)
             }
+            Value::Float(Some(f)) => {
+                sanitize_f32(f, "Int")?;
+
+                PrismaValue::Int(big_decimal_to_i64(BigDecimal::from_f32(f).unwrap(), "Int")?)
+            }
+            Value::Double(Some(f)) => {
+                sanitize_f64(f, "Int")?;
+
+                PrismaValue::Int(big_decimal_to_i64(BigDecimal::from_f64(f).unwrap(), "Int")?)
+            }
+            Value::Numeric(Some(dec)) => PrismaValue::Int(big_decimal_to_i64(dec, "Int")?),
+            other => to_prisma_value(other)?,
+        },
+        TypeIdentifier::BigInt => match p_value {
+            value if value.is_null() => PrismaValue::Null,
+            Value::Int32(Some(i)) => PrismaValue::BigInt(i as i64),
+            Value::Int64(Some(i)) => PrismaValue::BigInt(i),
+            Value::Bytes(Some(bytes)) => PrismaValue::BigInt(interpret_bytes_as_i64(&bytes)),
+            Value::Text(Some(ref txt)) => {
+                PrismaValue::BigInt(i64::from_str(txt.trim_start_matches('\0')).map_err(|_| create_error(&p_value))?)
+            }
+            Value::Float(Some(f)) => {
+                sanitize_f32(f, "BigInt")?;
+
+                PrismaValue::BigInt(big_decimal_to_i64(BigDecimal::from_f32(f).unwrap(), "BigInt")?)
+            }
+            Value::Double(Some(f)) => {
+                sanitize_f64(f, "BigInt")?;
+
+                PrismaValue::BigInt(big_decimal_to_i64(BigDecimal::from_f64(f).unwrap(), "BigInt")?)
+            }
```

Comment:
We can most likely cast directly from `f32`/`f64` to `i64`, but I'm fairly worried about how precision loss is handled compared to a `BigDecimal `-> `i64` coercion. I kept it that way to match the coercion previously made at the serialization layer to prevent any breaking changes.